client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FC, useEffect, useState } from 'react';
+import { StandardTextFieldProps } from '@common/components';
+import { BasicTextInput } from './BasicTextInput';
+import { NumUtils, RegexUtils } from '@common/utils';
+import { useFormatNumber, useCurrency } from '@common/intl';
+export interface NumericTextInputProps
+  extends Omit<StandardTextFieldProps, 'onChange'> {
+  onChange?: (value: number | undefined) => void;
+  width?: number | string;
+  defaultValue?: number;
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  decimalLimit?: number;
+  step?: number;
+  multiplier?: number;
+  value?: number | undefined;
+}
+
+export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
+  (
+    {
+      sx,
+      InputProps,
+      width = 75,
+      onChange = () => {},
+      defaultValue,
+      allowNegative,
+      min = allowNegative ? -NumUtils.MAX_SAFE_API_INTEGER : 0,
+      max = NumUtils.MAX_SAFE_API_INTEGER,
+      decimalLimit = 0,
+      step = 1,
+      multiplier = 10,
+      value,
+      ...props
+    },
+    ref
+  ) => {
+    const { format, parse } = useFormatNumber();
+    const {
+      options: { separator, decimal },
+    } = useCurrency();
+    const [textValue, setTextValue] = useState(format(value ?? defaultValue));
+
+    useEffect(() => {
+      setTextValue(format(value));
+    }, [value]);
```

Comment:
So we manage two representations of the number: `value`, which is passed in as a prop and updated via `onChange()` is the actual number, `textValue` is the current formatted and localised string representation.

I wouldn't have used an additional `useState` value for the string representation -- we should always be able to derive it from `value`. However, we need to manage it separately, since user needs to be able to type "-" and "." even though that's not enough to update the number yet, so `textValue` will be set independently to `onChange` when this happens.